builtin/credential/aws/path_login.go

```diff
@@ -1407,7 +1406,38 @@ func parseGetCallerIdentityResponse(response string) (GetCallerIdentityResponse,
 	return result, err
 }
 
-func submitCallerIdentityRequest(method, endpoint string, parsedUrl *url.URL, body string, headers http.Header) (*GetCallerIdentityResult, error) {
+func parseIamRequestHeaders(headersB64 string) (*http.Header, error) {
```

Comment:
`http.Header` is a typedef for `map[string][]string` and as such it's a reference type -- return it directly, not as a pointer.